app/models/chargeback_rate.rb

```diff
@@ -11,6 +20,8 @@ class ChargebackRate < ApplicationRecord
   validates_uniqueness_of   :guid
   validates_uniqueness_of   :description, :scope => :rate_type
 
+  virtual_column :assigned?, :type => :boolean
```

Comment:
@tenderlove @matthewd  is there a standard Rails pattern for this type of thing?  The idea is that the model knows whether it can destroy itself with helper methods that the UI can call to validate when and why deletion is not possible.  Then, `before_destroy` in this case becomes simplified to:
```ruby
before_destroy :my_before_destroy

def cannot_destroy?
  assigned?
end

def cannot_destroy_reason
  "rate is assigned and cannot be deleted"
end

def my_before_destroy
  if cannot_destroy?
    errors.add(:rate, cannot_destroy_reason)
    return false
  end
end
```

Now the UI can call the `cannot_destroy?` and `cannot_destroy_reason` methods, if it needs/wants to.